src/System.IO.Pipes/src/System/IO/Pipes/PipeStream.Unix.cs

```diff
@@ -58,6 +59,15 @@ internal static string GetPipePath(string serverName, string pipeName)
             // naming scheme used as that breaks the ability for code running on an older
             // runtime to connect to code running on the newer runtime.  That means we're stuck
             // with a tmp file for the lifetime of the server stream.
+            if (isCurrentUserOnly)
+            {
+                string directory = Path.Combine(Path.GetTempPath(), ".NET" + Interop.Sys.GetEUid());
```

Comment:
Nit: probably just ignore it, but... I know that there is a limit on the length here but in general we prefer predictable things, e.g.: we always use n chars, no matter the value of the id. An hypothetical corner case: same box an app uses a predefined pipe name, it works for one user but not for other. That said I am not sure here since something like `Interop.Sys.GetEUid().ToString("X8")` seems too wasteful while `ToString("X4")` will suffer the same inconsistency problem if ids get larger than 0x10000... (😛 if we just had code to convert to arbitrary bases... no, not adding it for this...)

Reply:
That's what @danmosemsft and I discussed yesterday offline, tried to figure out a way to do it with him but we got to the same conclusion :(